sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java

```diff
@@ -579,6 +579,16 @@ public boolean isSupertypeOf(TypeName other) {
     /** Convert the Java type used by the base {@link FieldType} to the input type. */
     @NonNull
     InputT toInputType(@NonNull BaseT base);
+
+    /** Convert the Java type used by the base {@link FieldType} to the input type. */
+    default BaseT toBaseTypeGeneric(Object input) {
+      return toBaseType((InputT) input);
+    }
+
+    /** Convert the Java type used by the base {@link FieldType} to the input type. */
+    default InputT toInputTypeGeneric(Object base) {
+      return toInputType((BaseT) base);
+    }
```

Comment:
Is there a way we could do this without adding these helpers? I'd rather keep the type erasure confined in `SchemaTranslation`

Reply:
Got something working.